routes/drones.js

```diff
@@ -1,37 +1,54 @@
 const express = require('express');
-
-// require the Drone model here
+const Drone = require("../models/Drone.model.js");
 
 const router = express.Router();
 
 router.get('/drones', (req, res, next) => {
-  // Iteration #2: List the drones
-  // ... your code here
-});
-
-router.get('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  Drone.find()
+  .then(allDronesFromDB =>{
+    res.render("drones-list", {drones: allDronesFromDB })
+  })
+  .catch(err => console.log(`The error ${err} came up!`))
+})
+
+router.get('/drones/create', (req, res) => {
+  res.render("drone-create")
 });
 
 router.post('/drones/create', (req, res, next) => {
-  // Iteration #3: Add a new drone
-  // ... your code here
+  const { name , propellers, maxSpeed } = req.body;
+
+  Drone.create({ name , propellers, maxSpeed })
+  .then(() => res.redirect("/drones"))
+  .catch(error => `Error occured while creating drones: ${error}`)
 });
 
 router.get('/drones/:id/edit', (req, res, next) => {
-  // Iteration #4: Update the drone
-  // ... your code here
+  const { id } = req.params;
+
+  Drone.findById ( id )
+  .then( droneToEdit => {
+    res.render("update-form", droneToEdit)
+  })
```

Comment:
You should always have a `.catch()` after your last `.then()` for handling/catching any errors that might occur.
```
.catch(error => console.log(`Error occured while creating drones: ${error}`))